src/arch/x86_64/Emit.zig

```diff
@@ -234,6 +235,16 @@ fn fixupRelocs(emit: *Emit) InnerError!void {
     }
 }
 
+fn mirUndefinedInstruction(emit: *Emit, inst: Mir.Inst.Index) InnerError!void {
+    const tag = emit.mir.instructions.items(.tag)[inst];
+    assert(tag == .ud);
+    const ops = emit.mir.instructions.items(.ops)[inst].decode();
+    switch (ops.flags) {
+        0b00 => return lowerToZoEnc(.ud2, emit.code),
+        else => unreachable,
+    }
+}
```

Comment:
As far as I know, `UD2` is always encoded as NP (meaning it will never take any operands) therefore it would be cleaner to follow the `mirSyscall` approach, i.e.:

```suggestion
    return lowerToZoEnc(.ud2, emit.code);
}
```

Reply:
Even the assert on the tag could be skipped IMHO.

Reply:
I like that! And it actually matches the way `mirSyscall` works, too (it doesn't take an `inst` either).

Reply:
Yep. There are some variations but not available on x86_64 as far as I can tell so we can always assume a single encoding for `UD2`. I will be rewriting the emitter and MIR real soon too when I start downstreaming the encoder library I am working on on the side, but in the meantime I think this is the right call to make.